pkg/provider/prefix/provider.go

```diff
@@ -334,8 +334,12 @@ func (p *ipv4PrefixProvider) GetPool(nodeName string) (pool.Pool, bool) {
 }
 
 func (p *ipv4PrefixProvider) IsInstanceSupported(instance ec2.EC2Instance) bool {
-	//TODO add check for Nitro instances
-	if instance.Os() == config.OSWindows {
+	limits, found := vpc.Limits[instance.Type()]
+	if !found {
+		return false
+	}
+
+	if instance.Os() == config.OSWindows && (limits.IsBareMetal || limits.Hypervisor == "nitro") {
 		return true
 	}
 
```

Comment:
Can you emit an event for cases of not found and not supported?

Reply:
Yea sure, I've added node events for both cases.

Reply:
Thanks, @jiechen0826. Please also make sure these user faced messages are tested (positive and negative).